### PR TITLE
Hiro clarinet updates 110722

### PR DESCRIPTION
**Pull request by trichards-hiro**
## Description

Added three new pages to the Hiro docs for new Clarinet features.

### CODE REVIEW

File: docs/clarinet/feature-guides/chainhooks.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+- portability and performance
+
+Hiro designed the event observer as a library, choosing to embed the library in Clarinet so you can run it on your local machine. You may also execute the library on the server side and then propagate these HTTP events to your other components.
+
+- correctness
+
+Due to the fact that blockchains can be forked, transactions can be discarded, and transactions can be added to a distinctly separate fork, ensuring correctness can be a challenge. There are many different ways you can end up with a state slightly differing from the canonical state, which is why correctness is an inherent limitation of chainhooks. 
+
+## Using Chainhoolks
```
Comment (lgalabru) [inline docs/clarinet/feature-guides/chainhooks.md:21]:
```suggestion
## Using Chainhooks
```

**Comment by lgalabru** [comment]
I added @mefrem who has a good handle on these features, and could help completing.

File: docs/clarinet/feature-guides/clarinet-deploy.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+## Overview
+
+Deployment plans allow teams to work together when deploying smart contracts. Hiro has created a set of primitives that will enable teams to work together more effectively, ensuring that these teams can create smart contracts faster and easier.
+
+By using Deployment Plans, you can simplify the smart contract deployment process to stacks or bitcoin environment.
+
+## Design
+
+Deployment Plans use a standardized protocol when deploying smart contracts on any network based on stacks or bitcoin (devnet, testnet, mainnet). They also minimize the complexity with deployments, making it much easier to deploy a contract without errors. Some of these complexities can include dependencies, chaining limits, the process to initialize a smart contract, and underlying deployment costs.
```
Comment (lgalabru) [inline docs/clarinet/feature-guides/clarinet-deploy.md:13]: How about:
Deployment Plans are an attempt to standardized protocols and contracts deployment, ...

File: docs/clarinet/feature-guides/clarinet-deploy.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+You can commit, audit, and test contracts without including any secrets in the Deployment Plan, and share these contracts without exposing any sensitive information.
+
+## Deployment Plans Primitives
+
+Deployment plans consist of the following four primitives:
+
+- deploy contracts
+- call contracts
+- send bitcoin transactions
+- wait for block
```
Comment (lgalabru) [inline docs/clarinet/feature-guides/clarinet-deploy.md:24]: - send stacks to an address or contract

Comment (lgalabru) [reply docs/clarinet/feature-guides/clarinet-deploy.md:24]: (recent addition)

File: docs/clarinet/feature-guides/clarinet-deploy.md (hunk 2 above)
Comment (mefrem) [reply docs/clarinet/feature-guides/clarinet-deploy.md:13]: How about:

Deployment Plans are **reproducible deployment steps that publish a “protocol”**—a collection of on-chain transactions and contracts—**to a network**, whether a local developer network, the public testnet, or into production on mainnet.

A deployment plan comprises the
1) smart contracts
2) accounts with their token balances, and 
3) content and sequence of transactions (across a single or multiple Stacks or Bitcoin blocks)

A Deployment Plan’s specifications exist on two files within a Clarinet project—the `Clarinet.toml` and network’s `.toml` file (e.g. `devnet.toml`) under the “deployments” folder

Comment (mefrem) [reply docs/clarinet/feature-guides/clarinet-deploy.md:13]: ^^ And this is to mix in/enhance @trichards-hiro's language as written, not replace.

File: docs/clarinet/feature-guides/chainhooks.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+Chainhooks were designed with a very specific set of requirements and limitations to make them easy to work with in a development environment. These constraints include:
+
+- portability and performance
+
+Hiro designed the event observer as a library, choosing to embed the library in Clarinet so you can run it on your local machine. You may also execute the library on the server side and then propagate these HTTP events to your other components.
+
+- correctness
+
+Due to the fact that blockcahins can be forked, transactions can be discarded, and transactions can be added to a distinctly separate fork, ensuring correctness is a challenge. There are many different ways you can end up with a state slightly differing from the canonical state, which is why correctness is an inherent limitation of chainhooks.
```
Comment (mefrem) [inline docs/clarinet/feature-guides/chainhooks.md]: Due to the fact that blockchains can be forked—and some period of uncertainty may arise as to which chain tip will assert itself as the canonical chain—there are challenges to guaranteeing the validity of a triggered predicate.

File: docs/clarinet/feature-guides/clarinet-deploy.md (hunk 2 above)
Comment (trichards-hiro) [reply docs/clarinet/feature-guides/clarinet-deploy.md]: incorporated this feedback into the content

File: docs/clarinet/feature-guides/clarinet-deploy.md (hunk 3 above)
Comment (trichards-hiro) [reply docs/clarinet/feature-guides/clarinet-deploy.md:34]: added to the documentation

File: docs/clarinet/feature-guides/chainhooks.md (hunk 4 above)
Comment (trichards-hiro) [reply docs/clarinet/feature-guides/chainhooks.md]: thanks! added to the content

File: docs/clarinet/feature-guides/chainhooks.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+- correctness
+
+Due to the fact that blockchains can be forked, transactions can be discarded, and transactions can be added to a distinctly separate fork, ensuring correctness can be a challenge. There are many different ways you can end up with a state slightly differing from the canonical state, which is why correctness is an inherent limitation of chainhooks. 
+
+## Using Chainhoolks
+
+In terms of deployment lifecycle, you can start using chainhooks locally today, using the latest versions of Clarinet.
+
+You may also choose to deploy chainhooks in your own environment, although this requires you to be responsible for your own depolyment instead of relying on Hiro's default deployment architecture.
```
[the remark this answers is not in the document]
Comment (trichards-hiro) [reply docs/clarinet/feature-guides/chainhooks.md]: clarified the content in this section to include these concepts.

**Review by mefrem — APPROVED** [review]
These typo fixes look good

**Comment by trichards-hiro** [comment]
Thanks @mefrem. I am going to fix the broken links that are preventing a merge, and then I will be good to ping @lgalabru and @hugocaillard for one final review before publishing.

**Comment by LakshmiLavanyaKasturi** [comment]
@trichards-hiro: Are there any blockers to merging this PR? If the issue is only with links, can you work with @janniks and get this merged asap? Thanks!

**Pull request merged**